Author: Olivier Tilloy <olivier.tilloy@canonical.com>
Description: revert of 074fa51f85b15ad2e2d6c804a03825d48dbeb063, which requires Clang r313108 (unreleased yet)

Index: dev/build/config/compiler/BUILD.gn
===================================================================
--- dev.orig/build/config/compiler/BUILD.gn
+++ dev/build/config/compiler/BUILD.gn
@@ -488,16 +488,6 @@ config("compiler") {
     ldflags += [ "-Wl,-z,keep-text-section-prefix" ]
   }
 
-  if (is_clang && !is_nacl && !use_xcode_clang) {
-    cflags += [ "-fcrash-diagnostics-dir=" + clang_diagnostic_dir ]
-
-    cflags += [
-      # TODO(hans): Remove this once Clang generates better optimized debug info
-      # by default. https://crbug.com/765793
-      "-mllvm",
-      "-instcombine-lower-dbg-declare=0",
-    ]
-  }
 
   # C11/C++11 compiler flags setup.
   # ---------------------------
@@ -2304,14 +2294,6 @@ config("symbols") {
       cflags += [ "-g2" ]
     }
 
-    # TODO(https://crbug.com/1050118): Investigate missing debug info on mac.
-    if (is_clang && !is_nacl && !use_xcode_clang && !is_mac && !is_ios) {
-      cflags += [
-        "-Xclang",
-        "-debug-info-kind=constructor",
-      ]
-    }
-
     if (use_debug_fission && !is_nacl && !is_android) {
       # NOTE: Some Chrome OS builds globally set |use_debug_fission| to true,
       # but they also build some targets against Android toolchains which aren't
